BUILDING.md

```diff
@@ -0,0 +1,82 @@
+build to work on CI servers using newly created agents to run the build and we generally run it
+the same way on our own machines.
+
+The build shell script and build.cmd script are provided as an easy way to run the above commands.
+In addition to passing their arguments through to build.cake, they can supply added arguments
+through the CAKE_ARGS environment variable. The rest of this document will assume use of these commands.
+
+There is one case in which use of the CAKE_ARGS environment variable will be essential, if not necessary.
+If you are running builds on a 32-bit Windows system, you must always supply the -Experimental argument
+to the build. Use set CAKE_ARGS=-Experimental to ensure this is always done and avoid having to type
+it out each time.
+
+Key arguments to build.cmd / build:
+ * -Target, -t <task>                 The task to run - see below.
+ * -Configuration, -c [Release|Debug] The configuration to use (default is Release)
+ * -Experimental, -e                  Use the experimental build of Roslyn
+
+The build.cake script contains a large number of interdependent tasks. The most 
+important top-level tasks to use are listed here:
+ * Build               Builds everything. This is the default if no target is given.
+ * Rebuild             Cleans the output directory and builds everything
+ * Test                Runs all tests. Dependent on Build.
+ * TestAll             Runs all tests. Dependent on Build.
+ * TestAllFrameworks   Runs all framework tests. Dependent on Build.
+ * Test45              Tests the 4.5 framework without building first.
+ * Test40              Tests the 4.0 framework without building first.
+ * Test20              Tests the 2.0 framework without building first.
+ * TestPortable        Tests the portable framework without building first.
+ * TestSL              Tests the Silverlight framework without building first.
+ * TestCF              Tests the compact framework without building first.
+ * TestEngine          Runs all engine tests. Dependent on Build.
+ * TestConsole         Runs the console tests. Dependent on Build.
+ * Package             Creates all packages without building first. See Note below.
+
+### Notes:
+ 1. By design, the Package target does not depend on Build. This is to allow re-packaging
+    when necessary without changing the binaries themselves. Of course, this means that
+    you have to be very careful that the build is up to date before packaging.
+
+ 2. If the compact framework is not installed on a machine, building and testing tasks
+    for CF are skipped and a warning is issued.
+
+ 3. Currently, Silverlight must be installed or an error occurs when trying to build or
+    test it. We will be changing this soon to match the strategy for CF.
+
+ 4. The Test and TestAll targets currently due the same thing. Both are retained for backward
+    backward compatibility and the semantics of at least one of them may change in the future.
```

Comment:
Have just seen this on a pc and it makes much more sense - the line breaks on mobile didn't match up to show the sentence split over two lines! Cheers!